packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -106,12 +108,38 @@ export interface ApplicationListenerProps extends BaseApplicationListenerProps {
   readonly loadBalancer: IApplicationLoadBalancer;
 }
 
+/**
+ * Options for ApplicationListener lookup
+ */
+export interface ApplicationListenerLookupOptions extends BaseListenerLookupOptions {
+  /**
+   * Filter listeners by listener protocol
+   * @default - does not filter by listener protocol
+   */
+  readonly listenerProtocol?: ApplicationProtocol;
+}
+
 /**
  * Define an ApplicationListener
  *
  * @resource AWS::ElasticLoadBalancingV2::Listener
  */
 export class ApplicationListener extends BaseListener implements IApplicationListener {
+  /**
+   * Look up an ApplicationListener.
+   */
+  static fromLookup(scope: Construct, id: string, options: ApplicationListenerLookupOptions): IApplicationListener {
+    const props = BaseListener._queryContextProvider(scope, {
+      userOptions: options,
+      loadBalancerType: cxschema.LoadBalancerType.APPLICATION,
+      listenerProtocol: options.listenerProtocol == ApplicationProtocol.HTTP
```

Comment:
```suggestion
      listenerProtocol: options.listenerProtocol === ApplicationProtocol.HTTP
```

Please use `===` and `!==` throughout.